cpp/src/io/parquet/page_delta_decode.cu

```diff
@@ -154,6 +424,159 @@ __global__ void __launch_bounds__(96)
   if (t == 0 and s->error != 0) { set_error(s->error, error_code); }
 }
 
+// Decode page data that is DELTA_BYTE_ARRAY packed. This encoding consists of a DELTA_BINARY_PACKED
+// array of prefix lengths, followed by a DELTA_BINARY_PACKED array of suffix lengths, followed by
+// the suffixes (technically the suffixes are DELTA_LENGTH_BYTE_ARRAY encoded). The latter two can
+// be used to create an offsets array for the suffix data, but then this needs to be combined with
+// the prefix lengths to do the final decode for each value. Because the lengths of the prefixes and
+// suffixes are not encoded in the header, we're going to have to first do a quick pass through them
+// to find the start/end of each structure.
+template <typename level_t>
+__global__ void __launch_bounds__(decode_block_size)
```

Comment:
What is the relationship between this and the code in page_string_decode.cu?  Is DELTA_BYTE_ARRAY always treated as strings and then the code in page_string_decode.cu is just for the preprocess/size computation?

Reply:
DELTA_BYTE_ARRAY is just another encoding for strings, joining PLAIN and DICTIONARY, and soon to be joined by DELTA_LENGTH_BYTE_ARRAY.  So yes, the code in page_string_decode.cu is preprocessing that's to some extent common to all the string decoders (which is why some DELTA stuff wound up in there rather than all living here).